File: src/data/chantTracks.ts
export type ChantId = '10000-reasons' | '10000-reasons-african' | 'be-still-my-soul' | 'soul-of-jesus-sanctify-me' | 'oceans';

export type ChantTrack = {
  id: ChantId;
  label: string;
  voiceKey?: string; // SoundKey
  instrumentalKey?: string; // SoundKey
  cues: string[];
  pauseDurationMs: number;
};

export const CHANT_TRACKS: Record<ChantId, ChantTrack> = {
  '10000-reasons': {
    id: '10000-reasons',
    label: '10,000 Reasons',
    voiceKey: 'db/10000_reasons.mp3',
    instrumentalKey: 'db/10000_reasons_instrumental.mp3',
    cues: ['Pause and thank God for today', 'Let praise rise with each breath'],
    pauseDurationMs: 12000,
  },
  '10000-reasons-african': {
    id: '10000-reasons-african',
    label: '10,000 Reasons (African)',
    voiceKey: 'db/10000_reasons_african.mp3',
    instrumentalKey: 'db/10000_reasons_instrumental.mp3',
    cues: ['Pause and thank God for today', 'Let praise echo with each breath'],
    pauseDurationMs: 12000,
  },
  'be-still-my-soul': {
    id: 'be-still-my-soul',
    label: 'Be Still My Soul',
    voiceKey: 'db/be_still_my_soul.mp3',
    instrumentalKey: 'db/be_still_my_soul_instrumental.mp3',
    cues: ['Rest your heart in God\'s care', 'Trust that Christ holds your future'],
    pauseDurationMs: 14000,
  },
  'soul-of-jesus-sanctify-me': {
    id: 'soul-of-jesus-sanctify-me',
    label: 'Soul of Jesus, Sanctify Me',
    voiceKey: 'db/anima_christi.mp3',
    instrumentalKey: 'db/anima_christi_instrumental.mp3',
    cues: ['Invite Jesus to sanctify you today', 'Be reassured in His saving mercy'],
    pauseDurationMs: 14000,
  },
  oceans: {
    id: 'oceans',
    label: 'Oceans (Spirit Lead Me)',
    voiceKey: 'db/oceans_voice.mp3',
    instrumentalKey: 'db/oceans_instrumental.mp3',
    cues: ['Notice where God invites you to trust deeper', 'Trust in every breath'],
    pauseDurationMs: 13000,
  },
};

export const getChantById = (id?: string | null): ChantTrack | null => {
  if (!id) return null;
  const key = id as ChantId;
  return (CHANT_TRACKS as any)[key] || null;
};

File: lib/features/meditation/domain/models/chant_tracks.dart
enum ChantId {
  tenThousandReasons('10000-reasons'),
  tenThousandReasonsAfrican('10000-reasons-african'),
  soulOfJesusSanctifyMe('soul-of-jesus-sanctify-me'),
  oceans('oceans');

  const ChantId(this.value);
  final String value;
}

class ChantTrack {
  final ChantId id;
  final String label;
  final String? voiceKey;
  final String? instrumentalKey;
  final List<String> cues;
  final int pauseDurationMs;

  const ChantTrack({
    required this.id,
    required this.label,
    this.voiceKey,
    this.instrumentalKey,
    required this.cues,
    required this.pauseDurationMs,
  });
}

class ChantTracks {
  static const Map<ChantId, ChantTrack> tracks = {
    ChantId.tenThousandReasons: ChantTrack(
      id: ChantId.tenThousandReasons,
      label: '10,000 Reasons',
      voiceKey: 'https://elbiblio.com/sounds/chants/10000_reasons.mp3',
      instrumentalKey: 'https://elbiblio.com/sounds/chants/10000_reasons_instrumental.mp3',
      cues: ['Pause and thank God for today', 'Let praise rise with each breath'],
      pauseDurationMs: 12000,
    ),
    ChantId.tenThousandReasonsAfrican: ChantTrack(
      id: ChantId.tenThousandReasonsAfrican,
      label: '10,000 Reasons (African)',
      voiceKey: 'https://elbiblio.com/sounds/chants/10000_reasons_african.mp3',
      instrumentalKey: 'https://elbiblio.com/sounds/chants/10000_reasons_instrumental.mp3',
      cues: ['Pause and thank God for today', 'Let praise echo with each breath'],
      pauseDurationMs: 12000,
    ),
    ChantId.soulOfJesusSanctifyMe: ChantTrack(
      id: ChantId.soulOfJesusSanctifyMe,
      label: 'Soul of Jesus, Sanctify Me',
      voiceKey: 'https://elbiblio.com/sounds/chants/anima_christi.mp3',
      instrumentalKey: 'https://elbiblio.com/sounds/chants/anima_christi_instrumental.mp3',
      cues: ['Invite Jesus to sanctify you today', 'Be reassured in His saving mercy'],
      pauseDurationMs: 14000,
    ),
    ChantId.oceans: ChantTrack(
      id: ChantId.oceans,
      label: 'Oceans (Spirit Lead Me)',
      voiceKey: 'https://elbiblio.com/sounds/chants/oceans_voice.mp3',
      instrumentalKey: 'https://elbiblio.com/sounds/chants/oceans_instrumental.mp3',
      cues: ['Notice where God invites you to trust deeper', 'Trust in every breath'],
      pauseDurationMs: 13000,
    ),
  };

  static List<ChantTrack> get allTracks => tracks.values.toList();

  static ChantTrack? getById(String? id) {
    if (id == null) return null;
    try {
      final chantId = ChantId.values.firstWhere((e) => e.value == id);
      return tracks[chantId];
    } catch (e) {
      return null;
    }
  }
}
